### PR TITLE
[CIS-278] Split Xcode project for v2 and v3

### PR DESCRIPTION
**Pull request by buh**
**Changes for v2**
- Removed _Realm_ target and related code from `Sources`
- Renamed `Example` folder to `Sample`

**Changes for v3**
- Added a new project `StreamChat_v3`
- SDK code moved from `StreamChatClient_v3` to `Sources_v3`
- Tests resources moved from `TestResources_v3` to `Tests_v3`
- Sample app moved from `V3SampleApp` to `Sample_v3`
- Test plans moved:
  - from `./StreamChatClient_v3.xctestplan` and `./StreamChatClientStress_v3.xctestplan`
  - to `Tests_v3/StreamChatClientTestPlan.xctestplan` and `Tests_v3/StreamChatClientStressTestPlan.xctestplan`

**Others**
- fixed codecov
- fixed fastlane
- hooks/pre-commit.sh
- `.gitignore` for `Sample/Carthage`

#skip_changelog

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -204,7 +204,7 @@ lane :test_carthage_integration do
   carthage_bootstrap
 
   scan(
-    project: "./Example/Carthage/ChatExample.xcodeproj",
+    project: "../Sample/Carthage/ChatExample.xcodeproj",
```
Comment (VojtaStavik) [inline fastlane/Fastfile:207]: Isn't the two `..` here a typo @buh ?

Comment (buh) [reply fastlane/Fastfile:207]: It should be a single dot there, but it doesn't work, I want to see a new error message.

Comment (buh) [reply fastlane/Fastfile:207]: It works locally with a single dot, but doesn't on GitHub.

File: Sample_v3/AppDelegate.swift [hunk 2]
```diff
@@ -7,7 +7,7 @@
 //
 
 import UIKit
-@testable import StreamChatClient_v3
+@testable import StreamChatClient
```
Comment (VojtaStavik) [inline Sample_v3/AppDelegate.swift:10]: Wait, this is not correct. We didn't change the name of the project, did we?? @buh

Comment (buh) [reply Sample_v3/AppDelegate.swift:10]: It’s a new project the target count as a final name. Then on release we will only rename project itself.

**Pull request merged**